source/simulator/assembly.cc

```diff
@@ -1312,14 +1369,21 @@ namespace aspect
                           std_cxx11::_1),
          internal::Assembly::Scratch::
          StokesSystem<dim> (finite_element, mapping, quadrature_formula,
-                            (update_values    | update_gradients |
+                            face_quadrature_formula,
+                            (update_values    |
+                             update_gradients |
                              update_quadrature_points  |
-                             update_JxW_values |
-                             (rebuild_stokes_matrix == true
-                              ?
-                              update_gradients
-                              :
-                              UpdateFlags(0))),
```

Comment:
@gassmoeller : If you look at the previous code, you'll see that we *always* added the gradients, and then sometimes just added it a second time again. We just simplified the code.